tree_node.go

```diff
@@ -34,7 +33,7 @@ type TreeNode struct {
 	path         uint64
 	depth        uint8
 	hasher       *Hasher
-	extended     bool
+	temporary    bool
```

Comment:
How can I understand `temporary`? What is the motivation to rename it?

Reply:
The temporary nodes without child data.
will be extended when it needs to be searched down.